proj Плотников, Курагин

OFFdevil opened the pull request:
Плотников: выполнил свою часть проекта

OFFdevil committed:
version 1.0.0

kajigor commented:
Кажется, возникло недопонимание по поводу синтаксиса языка. 
* Унифицировать можно 2 атома, не только переменные. 
* Присвоений и объявлений переменных в абстрактном синтаксисе нет, во что вы собираетесь их парсить? 
* Не поняла, что за проблемы с комментариями.

OFFdevil commented:
1) Был изменён синтаксис языка
2) Унификация теперь возможна для атомов
3) Исправлен парсер(неправильно строил дерево)
4) Обновлены тесты

kajigor commented:
* Унификация это одна операция (такое чуть более мудрое сравнение на равенство), в абстрактном синтаксисе языка нет ничего для других бинарных операторов, которые вы используете. 
* Откуда все-таки ограничения на пробелы после аргументов в объявлении функций? 
* Проверка уникальности аргументов и имен функций -- это все не имеет отношения к синтаксическому анализу и не должно выполняться на уровне парсера. 
* Можете поподробнее описать значимость пробелов в вашем языке?

OFFdevil commented:
1. Fixed
2. Не помню почему именно у меня не получилось сделать так, чтобы можно было пробелы ставить, но после 1 часа придумывания решения, я решил, что можно в особенности синтаксиса это добавить, что нельзя пробелы ставить
3. Fixed
4. Они определяют вложенность функций и их выполнение. Функции выполняются снизу вверх, но если после функции с количеством пробелов = 2, стоит функция с количеством пробелов = 3, то сначала выполнится вторая, а только потом первая. То есть сначала выполняются функции в вложенном блоке, а только потом функция, которая стоит перед этим вложенным блоком.

kajigor commented:
Плотников: думаю, пойдет. Надо будет сделать презентацию.
